pyannote/audio/utils/powerset.py

```diff
@@ -138,3 +138,59 @@ def to_powerset(self, multilabel: torch.Tensor) -> torch.Tensor:
             torch.argmax(torch.matmul(multilabel, self.mapping.T), dim=-1),
             num_classes=self.num_powerset_classes,
         )
+
+    def permutation_powerset(self, permutation_ml: torch.Tensor) -> torch.Tensor:
+        """Find the equivalent to a multilabel permutation in the powerset class space.
+
+        Parameters
+        ----------
+        permutation_ml: torch.Tensor
+            A multilabel permutation(s), (num_classes,) or (batch, num_classes)-shaped.
+
+        Returns
+        -------
+        torch.Tensor
+            The corresponding powerset permutation(s), (num_powerset_classes,)
+            or (batch, num_powerset_classes)-shaped depending on the input.
+
+        Example
+        ---------
+        With Powerset(2,2), the powerset classes are [nonspeech, spk1, spk2, spk1+spk2],
+        while the multilabel 'classes' are [spk1, spk2].
+        If we get for the equivalent to the multilabel permutation [1,0] (=[spk2, spk1]):
+
+        >>> powerset = Powerset(num_classes=2, max_set_size=2)
+        >>> permutation_ml = torch.tensor([1, 0])
+        >>> permutation_ps = powerset.permutation_powerset(permutation_ml)
+        >>> permutation_ps
+        tensor([0, 2, 1, 3])
+
+        We obtain the powerset permutation with classes spk1 and spk2 permutated.
+        """
+
+        def find_ps_perm(ps, perm_ml):
```

Comment:
Can't you make it a method of `Powerset`?

Reply:
Done ! Not sure about the naming though

Reply:
I usually call this kind of helper function the same name as the same one but prepended by `_`.

Reply:
Just renamed it :+1: